install_files/ansible-base/roles/noble-migration/tasks/main.yml

```diff
@@ -50,6 +80,29 @@
         delay: 10
         timeout: 300
 
+    - name: Recheck host's migration state
+      ansible.builtin.slurp:
+        src: /etc/securedrop-noble-migration-state.json
+      register: migration_json
+
+    - name: Extract current state
+      set_fact:
+        # slurp base64-encodes our file
+        finished_state: "{{ (migration_json.content | b64decode | from_json)['finished'] }}"
+
+    - name: Extract current state (2)
+      set_fact:
+        finished_state_index: "{{ stages.index(finished_state) }}"
+
+    - debug:
+        var: finished_state
+    - debug:
+        var: finished_state_index
+
+- name: Phase 2 of migration
+  # After PendingUpdates (index 1) but before Reboot (index 15)
+  when: finished_state_index|int >= 1 and finished_state_index|int < 15
```

Comment:
The `|int` filter is necessary because technically the index variable is a string, so ansible won't let you run `>=` with a integer.